workflow/schemas/resources.schema.yaml

```diff
@@ -2,144 +2,170 @@ $schema: "http://json-schema.org/draft-04/schema#"
 description: snakemake configuration file
 type: object
 properties:
-  bwa_mem:
+  default_resources:
     type: object
+    description: default resource definitions
     properties:
-      threads:
-        type: integer
-        description: number of threads that will be used by the 'bwa_mem' rule
-      time:
-        type: string
-        description: max execution time for the 'bwa_mem' rule
       mem_mb:
         type: integer
-        description: memory used for 'bwa_mem'
+        description: memory in MB used per cpu
       mem_per_cpu:
         type: integer
         description: memory used per cpu for 'bwa_mem'
       partition:
         type: string
-        description: partition to use on the cluster for 'bwa_mem'
+        description: partition to use on cluster
+      time:
+        type: string
+        description: max execution time
+      threads:
+        type: integer
+        description:  number of threads to be available
+    required:
+      - threads
+      - time
+      - mem_mb
+      - mem_per_cpu
+      - partition
 
```

Comment:
I removed the empty lines in between the different "paragraphs" but it doesn't really matter. It is just a consistency thing at best.

Reply:
We talked about it last week and me and the others found it easier to read with empty lines between.